WordPress/Classes/ViewRelated/Pages/PageListViewController.m

```diff
@@ -0,0 +1,531 @@
+    NSString *publishButtonTitle = NSLocalizedString(@"Publish Immediately", @"Label for a button that moves a page to the published folder, publishing with the current date/time.");
+    NSString *trashButtonTitle = NSLocalizedString(@"Move to Trash", @"Label for a button that moves a page to the trash folder");
+    NSString *cancelButtonTitle = NSLocalizedString(@"Cancel", @"Label for a cancel button");
+    NSString *deleteButtonTitle = NSLocalizedString(@"Delete Permanently", @"Label for a button permanently deletes a page.");
+
+    UIActionSheet *actionSheet;
+    PostListStatusFilter filter = [self currentPostListFilter].filterType;
+    if (filter == PostListStatusFilterTrashed) {
+        actionSheet = [[UIActionSheet alloc] initWithTitle:nil
+                                                   delegate:self
+                                          cancelButtonTitle:cancelButtonTitle
+                                     destructiveButtonTitle:nil
+                                          otherButtonTitles:publishButtonTitle, draftButtonTitle, deleteButtonTitle, nil];
+    } else if (filter == PostListStatusFilterPublished) {
+        actionSheet = [[UIActionSheet alloc] initWithTitle:nil
+                                                  delegate:self
+                                         cancelButtonTitle:cancelButtonTitle
+                                    destructiveButtonTitle:nil
+                                         otherButtonTitles:viewButtonTitle, draftButtonTitle, trashButtonTitle, nil];
+    } else {
+        // draft or scheduled
+        actionSheet = [[UIActionSheet alloc] initWithTitle:nil
+                                                  delegate:self
+                                         cancelButtonTitle:cancelButtonTitle
+                                    destructiveButtonTitle:nil
+                                         otherButtonTitles:viewButtonTitle, publishButtonTitle, trashButtonTitle, nil];
+    }
+    CGRect frame = CGRectZero;
+    frame.size = button.bounds.size;
+    [actionSheet showFromRect:frame inView:button animated:YES];
+}
+
+- (void)cell:(UITableViewCell *)cell receivedRestoreActionForProvider:(id<WPContentViewProvider>)contentProvider
+{
+    AbstractPost *apost = (AbstractPost *)contentProvider;
+    [self restorePost:apost];
+}
+
+
+#pragma mark - UIActionSheet Delegate Methods
+
+- (void)actionSheet:(UIActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex
+{
+    if (buttonIndex == actionSheet.cancelButtonIndex) {
+        return;
+    }
+
+    NSError *error;
+    Page *page = (Page *)[self.managedObjectContext existingObjectWithID:self.currentActionedPageObjectID error:&error];
+    if (error) {
+        DDLogError(@"%@, %@, %@", NSStringFromClass([self class]), NSStringFromSelector(_cmd), error);
+    }
+
+    PostListStatusFilter filter = [self currentPostListFilter].filterType;
+
+    if (buttonIndex == 0) {
```

Comment:
A bit torn about this.  Since we're dealing with indexes, and since button index is situational I'm not sure creating consts would make this much clearer...